File: crates/domain/src/catalog/series_detail.rs
use crate::catalog::Series;
use crate::metadata::{CreditedPerson, ExternalId, Extra, Genre, Rating, Studio};

#[derive(Debug, Clone)]
pub struct SeriesDetail {
    pub series: Series,
    pub genres: Vec<Genre>,
    pub credits: Vec<CreditedPerson>,
    pub studios: Vec<Studio>,
    pub ratings: Vec<Rating>,
    pub external_ids: Vec<ExternalId>,
    pub extras: Vec<Extra>,
    pub episodes_total: u32,
    pub episodes_with_available_version: u32,
}

File: clients/flutter/lib/model/catalog/series_detail.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:shadowmask/model/common/artwork.dart';
import 'package:shadowmask/model/common/content_rating.dart';
import 'package:shadowmask/model/catalog/detail_dimensions.dart';

part 'series_detail.freezed.dart';
part 'series_detail.g.dart';

@freezed
abstract class SeriesDetail with _$SeriesDetail {
  const factory SeriesDetail({
    required String id,
    required String title,
    int? year,
    String? overview,
    ContentRating? contentRating,
    @Default(false) bool manuallyEdited,
    String? addedAt,
    String? updatedAt,
    Artwork? artwork,
    @Default(<Genre>[]) List<Genre> genres,
    @Default(<Credit>[]) List<Credit> credits,
    @Default(<Studio>[]) List<Studio> studios,
    @Default(<Rating>[]) List<Rating> ratings,
    @Default(<ExternalId>[]) List<ExternalId> externalIds,
    @Default(<Extra>[]) List<Extra> extras,
    @Default(0) int episodesTotal,
    @Default(0) int episodesWithAvailableVersion,
  }) = _SeriesDetail;

  factory SeriesDetail.fromJson(Map<String, dynamic> json) =>
      _$SeriesDetailFromJson(json);
}
